apps/leo_gateway/src/leo_nfs_readdir_state_ets.erl

```diff
@@ -0,0 +1,130 @@
+%%======================================================================
+%%
+%% Leo Gateway
+%%
+%% Copyright (c) 2012-2015 Rakuten, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%======================================================================
+-module(leo_nfs_readdir_state_ets).
+
+-behaviour(leo_nfs_readdir_state_behaviour).
+-behaviour(gen_server).
+
+-include("leo_gateway.hrl").
+-include_lib("leo_logger/include/leo_logger.hrl").
+-include_lib("leo_commons/include/leo_commons.hrl").
+-include_lib("stdlib/include/ms_transform.hrl").
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+-export([info/0, start_link/1]).
+-export([add_readdir_entry/2, get_readdir_entry/1, del_readdir_entry/1]).
+
+-record(state, {
+          scan_interval :: integer(),
+          entry_ttl     :: integer()
+         }).
+
+
+-define(LEO_GW_NFS_READDIR_ENTRY_ETS_TBL, readdir_entry_ets_tbl).
+
+info() ->
+    ets:info(?LEO_GW_NFS_READDIR_ENTRY_ETS_TBL).
+
+
+start_link(Params) ->
+    ets:new(?LEO_GW_NFS_READDIR_ENTRY_ETS_TBL, [set, named_table, public]),
```

Comment:
~~As LEO_GW_NFS_READDIR_ENTRY_ETS_TBL is accessed by only this gen_server process,
Setting `private` rather than `public` would be appropriate.~~

Sorry I'm missing the table is actually accessed by any caller process spawned by nfs_rpc_server.